webapp-django/crashstats/signature/views.py

```diff
@@ -565,6 +568,23 @@ def signature_summary(request, params):
             reverse=True
         )
 
+    if 'android_cpu_abi' in facets:
+        mobile_devices = []
+
+        for cpu_abi in facets['android_cpu_abi']:
+            for manufacturer in cpu_abi['facets']['android_manufacturer']:
+                for model in manufacturer['facets']['android_model']:
+                    for version in model['facets']['android_version']:
+                        mobile_devices.append({
+                            'cpu_abi': cpu_abi['term'],
+                            'manufacturer': manufacturer['term'],
+                            'model': model['term'],
+                            'version': version['term'],
+                            'count': version['count'],
+                        })
+
+        facets['android_cpu_abi'] = mobile_devices
+
```

Comment:
I can't say anything wrong with it except that it looks tremendously complex. 4 nested for-loops is not something you see very often. 

I guess it's just the nature of ES and aggregations. Or possibly that I'm too used to SQL. 
Would there be any way we can abstract this so that view code isn't so deeply nested. 
In particular, considering that you might want to expose this to users; they'd appreciate a simpler output.